packages/task-release/release.js

```diff
@@ -0,0 +1,86 @@
+import { spawn as spawnCb } from "node:child_process";
+import { readFileSync } from "node:fs";
+import { cp, mkdir, readdir, rm, writeFile } from "node:fs/promises";
+import mainPackage from "../../package.json" assert { type: "json" };
+
+function spawn(command, args, options) {
+  return new Promise((resolve, reject) => {
+    const child = spawnCb(command, args, options);
+    child.on("exit", (code) => {
+      if (code === 0) {
+        return resolve();
+      }
+      reject(new Error(`Process exited with code ${code}`));
+    })
+  });
+}
+
+export const platforms = [
+  {
+    os: "darwin",
+    arch: "arm64",
+    exe: "./bin/task",
+  },
+  {
+    os: "darwin",
+    goarch: "amd64",
+    arch: "x64",
+    exe: "./bin/task",
+  },
+  {
+    os: "linux",
+    arch: "arm64",
+    exe: "./bin/task",
+  },
+  {
+    os: "linux",
+    goarch: "amd64",
+    arch: "x64",
+    exe: "./bin/task",
+  },
+  {
+    os: "win32",
+    goos: "windows",
+    goarch: "amd64",
+    arch: "x64",
+    exe: "./bin/task.exe",
+  },
+];
+
+const packageJson = readFileSync("package.json", "utf-8");
+
+async function createFolder(platform) {
+  const folder = `task-${platform.os}-${platform.arch}`
+
+  console.debug(`Creating folder ${folder}`)
+  await mkdir(`${folder}/bin`, { recursive: true })
+  console.debug(`Copying files to ${folder}`)
+  await writeFile(`${folder}/package.json`, packageJson.replace(/{platform}/g, platform.os).replace(/{arch}/g, platform.arch).replace(/{version}/g, mainPackage.version))
```

Comment:
Maybe use `.replaceAll('{platform}', ...)`? don't this we need regex here.

Reply:
What are the advantages of using replaceAll instead of replace with a global flag?

Reply:
readers don't need to parse regex in mind